### DISCUSSION TITLE
Error importing node set into asyncua server - Unknown datatype for field

### DISCUSSION THREAD

**Discussion by tiagodrm** [FreeOpcUa/opcua-asyncio#1248] [Q&A] [answered]
Hi. I´m using asyncua 1.0.1 on Python 3.10.4 and when trying to import the standard TMC node set (https://github.com/OPCFoundation/UA-Nodeset/tree/TMC-1.00-2017-10-11/TMC) using the example on server-example.py I get the following error:
ERROR:asyncua.common.structures104:Unknown datatype for field: StructureField(Name='Message', Description=LocalizedText(Locale=None, Text=''), DataType=NodeId(Identifier=3002, NamespaceIndex=4, NodeIdType=<NodeIdType.Numeric: 2>), ValueRank=1, ArrayDimensions=[0], MaxStringLength=0, IsOptional=False) in structure:MethodExecutionFeedbackType, please report
WARNING:asyncua.common.xmlimporter:failure adding node NodeData(nodeid:NodeId(Identifier=6668, NamespaceIndex=4, NodeIdType=<NodeIdType.Numeric: 2>)) Unknown datatype for field: StructureField(Name='Message', Description=LocalizedText(Locale=None, Text=''), DataType=NodeId(Identifier=3002, NamespaceIndex=4, NodeIdType=<NodeIdType.Numeric: 2>), ValueRank=1, ArrayDimensions=[0], MaxStringLength=0, IsOptional=False) in structure:MethodExecutionFeedbackType, please report
Traceback (most recent call last):
  File "/Users/tdmacedo/.virtualenvs/asyncua-test/lib/python3.10/site-packages/asyncua/common/xmlimporter.py", line 403, in _make_ext_obj
    extclass = self._get_ext_class(obj.objname)
  File "/Users/tdmacedo/.virtualenvs/asyncua-test/lib/python3.10/site-packages/asyncua/common/xmlimporter.py", line 398, in _get_ext_class
    raise Exception("Error no extension class registered ", name, nodeid)
Exception: ('Error no extension class registered ', 'MaterialType', NodeId(Identifier=3010, NamespaceIndex=4, NodeIdType=<NodeIdType.Numeric: 2>))

Any idea on what's going on?

**Comment by schroeder-** [comment, marked as answer]
Maybe there is a error in the old nodeset? 
Did you try to use the latest tmc? https://github.com/OPCFoundation/UA-Nodeset/tree/latest/TMC

Also with asyncua 1.0.1 you mean the current master from github?

**Reply by tiagodrm** [reply]
Hi @schroeder- and thanks for the quick reply.

I tested using the latest version of the nodeset and got the same error — just under a different node. In any case, the original nodeset is imported successfully in the Prosys simulator, so the issue seems to be opcua-asyncio specific.

By asyncua 1.0.1 I had meant the latest version from pypi, released in 11/7/22.
I've installed the current version from master (pip install git+https://github.com/FreeOpcUa/opcua-asyncio.git#egg=asyncua) and tried to import both the original xml and the latest. Unfortunately, the errors are the same weather I use opcua-asyncio from pypi or master.

I'm trying to import the original nodesets as such: 
await server.import_xml("Opc.Ua.Di.NodeSet2.xml")
await server.import_xml("Opc.Ua.TMC.NodeSet2.xml")

And for the latest nodesets I'm doing:
await server.import_xml("latest/Opc.Ua.Di.NodeSet2.xml")
await server.import_xml("latest/Opc.Ua.PackML.NodeSet2.xml")
await server.import_xml("latest/Opc.Ua.TMC.NodeSet2.xml")

In both cases the nodesets' dependencies are imported with no issues, but the TMC nodeset imports fail.

Am I missing something?

**Reply by schroeder-** [reply]
We have a problem with nodeset that contain Variables with Values. Our sorting algorithm doesn't check if the datatype of the values is load.

**Reply by schroeder-** [reply]
With #1259 i manged to load the companion specification. You have to remove the node with the id 8653 because there is a error in the nodeset.
